### ISSUE TITLE
RunPCA error 

### ISSUE THREAD

**Issue by upfriend** [satijalab/seurat#1525]
Hi, I followed your clustering tutorial (for a dataset of PBMC) w/ the new version of Seurat before starting w/ my 10X dataset, but found the following error when running RunPCA:

> pbmc <- RunPCA(pbmc, features = VariableFeatures(object = pbmc))
 Error in irlba(A = t(x = object), nv = npcs, ...) : 
  BLAS/LAPACK routine 'DLASCL' gave error code -4

Can you figure out what's the problem? I re-installed Seurat w/ the development version (v3.0.1) but the error still persisted. Any help will be appreciated.

Thanks,
Dong-Wook

**Comment by andrewwbutler** [comment]
Hi,

Are you able to run `irlba` on a similarly sized matrix (3000 x 2000)? 

```R
library(irlba)
mat <- matrix(data = rnorm(n = 3000*2000), nrow = 3000, ncol = 2000)
test <- irlba(mat, nv = 50)
```

**Comment by upfriend** [comment]
Hi, thanks for your reply. Btw, here's the result:

> library(irlba)
> mat <- matrix(data = rnorm(n = 3000*2000), nrow = 3000, ncol = 2000)
> test <- irlba(mat, nv = 50)
Error in irlba(mat, nv = 50) : 
  BLAS/LAPACK routine 'DLASCL' gave error code -4

I think there's something wrong in my R (I did re-install it but still same). Do you have any other idea?

Thanks!

**Comment by mojaveazure** [comment]
This looks like an issue with [irlba](https://github.com/bwlewis/irlba), I'd suggest bringing this up with them.

**Comment by gokceneraslan** [comment]
It's already reported: https://github.com/bwlewis/irlba/issues/14

**Comment by gokceneraslan** [comment]
It'd be nice to have an easy way to fall back to R's `svd()` or `prcomp()` methods in `RunPCA`.

**Comment by biotommy** [comment]
I was getting the same error until I tried turning the "approx" argument to FALSE:

obj <- RunPCA(obj, approx = FALSE)
